File: simple_pub_sub/src/sub.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void std_cb(const std_msgs::String::ConstPtr &msg) {
    ROS_INFO("i received: %s", msg->data.c_str());
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "sim_sub");
    ros::NodeHandle nh;

    ros::Subscriber std_sub = nh.subscribe("chatter", 1000, &std_cb);

    ros::spin();
    return 0;
}

File: pyhon_src/simple_pub_sub/scripts/sub.py
#! /usr/bin/env python3

import rospy
from std_msgs.msg import String

def msg_cb(msg):
    tmp_str = "%s: i received %s" %(rospy.get_caller_id(), msg.data)
    rospy.loginfo(tmp_str)

def main():
    rospy.init_node("sim_sub")

    rospy.Subscriber("chatter", String, msg_cb)

    rospy.spin()



if __name__ == "__main__":
    main()